opcua/common/instantiate.py

```diff
@@ -5,8 +5,42 @@
 
 from opcua import Node
 from opcua import ua
+from opcua.common.uaerrors import UaError
 
-
+#TODO: Should this be move to a more general location that instantiate.py ?
+def get_sub_types(server, node, includeitself=False, skipbase = True):
+    """
+    return get all subtype parents of node recursive
+    :param server: used in case node is nodeid         
+    :param node: can be a ua.Node or ua.NodeId
+    :param includeitself: include also node to the list
+    :param skipbase don't include the toplevel one
+    :returns list of ua.Node, top parent first 
+    """
+    parents =[]    
+    if isinstance(node, ua.NodeId): 
+        node = Node(server, node)    
+    if includeitself:
+        parents.append(node)
+    parents.extend(_get_sub_types(node))
+    if skipbase and len(parents) > 1:
+        parents = parents [:-1]
+        
+    return parents
+
+def _get_sub_types(node):
+    """
+    recursive implementation of get_sub_types
+    """
+    basetypes = []
+    parents = node.get_referenced_nodes(refs=ua.ObjectIds.HasSubtype, direction=ua.BrowseDirection.Inverse, includesubtypes=True)
+    if len(parents) != 0:  
+        #TODO: Is it possible to have multiple subtypes ? If so extended support for it
+       basetypes.append(parents[0])  
+       basetypes.extend( _get_sub_types(parents[0]) )
+       
+    return basetypes
+    
```

Comment:
@oroulet  I will move the function to ua_utils an rename it to `get_node_supertypes`. The name get_sub_types was misleading because it was about the super types and not the subtypes.

Concerning the merge/replace; this is no applicable due the difference in functionality. But I will move also the events.py `get_node_subtypes `to ua_utils.py.